src/composables/useImageUpload.ts

```diff
@@ -0,0 +1,55 @@
+import { ref } from 'vue';
+import { upload as pin } from '@snapshot-labs/pineapple';
+
+export function useImageUpload() {
+  const isUploadingImage = ref(false);
+  const imageUploadError = ref('');
+  const imageUrl = ref('');
+  const imageName = ref('');
+
+  const reset = () => {
+    isUploadingImage.value = false;
+    imageUploadError.value = '';
+    imageUrl.value = '';
+    imageName.value = '';
+  };
+
+  const upload = async (
+    file: File | undefined,
+    onSuccess: (image: { name: string; url: string }) => void
+  ) => {
+    reset();
+    if (!file) return;
+    isUploadingImage.value = true;
+    const formData = new FormData();
+
+    // TODO: Additional Validations - File Size, File Type, Empty File, Hidden File
+    if (!['image/jpeg', 'image/jpg', 'image/png'].includes(file.type)) {
+      imageUploadError.value = 'Unsupported file type';
+      isUploadingImage.value = false;
+      return;
+    }
+    formData.append('file', file);
```

Comment:
```suggestion
    // TODO: Additional Validations - File Size, File Type, Empty File, Hidden File
    if (!['image/jpeg', 'image/jpg', 'image/png'].includes(file.type)) {
      imageUploadError.value = 'Unsupported file type';
      isUploadingImage.value = false;
      return;
    }
    
    const formData = new FormData();
    formData.append('file', file);
```

`formData`'s initialization and set up looks weird being separated by unrelated function.